src/IdealizedMesoscaleExperiments/eddying_channel.jl

```diff
@@ -0,0 +1,301 @@
+using Statistics
+using Printf
+using Logging
+using JLD2
+using NCDatasets
+using GeoData
+using Oceanostics
+using Oceananigans
+using Oceananigans.Units
+
+using Oceanostics.TKEBudgetTerms: TurbulentKineticEnergy, ZShearProduction
+
+using LESbrary.Utils: SimulationProgressMessenger, fit_cubic, poly
+using LESbrary.NearSurfaceTurbulenceModels: SurfaceEnhancedModelConstant
+using LESbrary.TurbulenceStatistics: first_order_statistics,
+                                     first_through_second_order,
+                                     turbulent_kinetic_energy_budget,
+                                     subfilter_momentum_fluxes,
+                                     subfilter_tracer_fluxes,
+                                     ViscousDissipation
+
+try
+    using CairoMakie
+catch
+    using GLMakie
+finally
+    @warn "Could not load either CairoMakie or GLMakie; animations are not available."
+end
+
+Logging.global_logger(OceananigansLogger())
+
+@inline passive_tracer_forcing(x, y, z, t, p) = p.μ⁺ * exp(-(z - p.z₀)^2 / (2 * p.λ^2)) - p.μ⁻
+
+# # Code credit: https://discourse.julialang.org/t/collecting-all-output-from-shell-commands/15592
+# function execute(cmd::Cmd)
+#     out, err = Pipe(), Pipe()
+#     process = run(pipeline(ignorestatus(cmd), stdout=out, stderr=err))
+#     close(out.in)
+#     close(err.in)
+#     return (stdout = out |> read |> String, stderr = err |> read |> String, code = process.exitcode)
+# end
+
+function eddying_channel_simulation(; τ = 0.2, β = 1e-11, ridge_height = 0.0)
+    filepath = "tau_" * string(τ) * "_beta_" * string(β) * "_ridge_height_" * string(ridge_height)
+    filename = filepath
+    # Domain
+    Lx = 4000kilometers # zonal domain length [m]
+    Ly = 2000kilometers # meridional domain length [m]
+    Lz = 3kilometers    # depth [m]
+
+    # number of grid points
+    Nx = 80
+    Ny = 40
+    Nz = 30
+
+    save_fields_interval = 7days
+    stop_time = 20years + 1day
+    Δt₀ = 5minutes
+
+    grid = RectilinearGrid(arch;
+        topology = (Periodic, Bounded, Bounded),
+        size = (Nx, Ny, Nz),
+        halo = (3, 3, 3),
+        x = (0, Lx),
+        y = (0, Ly),
+        z = (-Lz, 0))
+
+    @info "Built a grid: $grid."
+
+    #####
+    ##### Boundary conditions
+    #####
+
+    α = 2e-4     # [K⁻¹] thermal expansion coefficient
+    g = 9.8061   # [m s⁻²] gravitational constant
+    cᵖ = 3994.0   # [J K⁻¹] heat capacity
+    ρ = 1024.0   # [kg m⁻³] reference density
```

Comment:
Let's avoid defining this parameter and use the "primitive" quantities: kinematic momentum flux, buoyancy flux, etc.